Source/Scene/ModelExperimental/ModelExperimentalUtility.js

```diff
@@ -0,0 +1,127 @@
+import Cartesian3 from "../../Core/Cartesian3.js";
+import defined from "../../Core/defined.js";
+import Matrix4 from "../../Core/Matrix4.js";
+import Quaternion from "../../Core/Quaternion.js";
+import RuntimeError from "../../Core/RuntimeError.js";
+import Axis from "../Axis.js";
+
+/**
+ * Utility functions for {@link ModelExperimental}.
+ *
+ * @private
+ */
+export default function ModelExperimentalUtility() {}
+
+/**
+ * Create a function for reporting when a model fails to load
+ *
+ * @param {ModelExperimental} model The model to report about
+ * @param {String} type The type of object to report about
+ * @param {String} path The URI of the model file
+ * @return {Function} An error function that
+ */
+ModelExperimentalUtility.getFailedLoadFunction = function (model, type, path) {
+  return function (error) {
+    var message = "Failed to load " + type + ": " + path;
+    if (defined(error)) {
+      message += "\n" + error.message;
+    }
+    model._readyPromise.reject(new RuntimeError(message));
+  };
+};
+
+/**
+ * Get a transformation matrix from a node in the model.
+ *
+ * @param {ModelComponents.Node} node The node components
+ * @return {Matrix4} The computed transformation matrix. If no transformation matrix or parameters are specified, this will be the identity matrix.
+ */
+ModelExperimentalUtility.getNodeTransform = function (node) {
+  if (defined(node.matrix)) {
+    return node.matrix;
+  }
+
+  return Matrix4.fromTranslationQuaternionRotationScale(
+    defined(node.translation) ? node.translation : new Cartesian3(),
+    defined(node.rotation) ? node.rotation : Quaternion.IDENTITY,
+    defined(node.scale) ? node.scale : new Cartesian3(1, 1, 1)
+  );
+};
+
+/**
+ * Find an attribute by semantic such as POSITION or TANGENT.
+ *
+ * @param {ModelComponents.Primitive} primitive The primitive components
+ * @param {VertexAttributeSemantic} semantic The semantic to search for
+ * @param {Number} setIndex The set index of the semantic. May be undefined for some semantics (POSITION, NORMAL, TRANSLATION, ROTATION, for example)
+ * @return {ModelComponents.Attribute} The selected. attribute, or undefined if not found.
+ */
+ModelExperimentalUtility.getAttributeBySemantic = function (
+  object,
+  semantic,
+  setIndex
+) {
+  var i;
+  var attributes = object.attributes;
+  var attributesLength = attributes.length;
+  for (i = 0; i < attributesLength; ++i) {
+    var attribute = attributes[i];
+    var matchesSetIndex = defined(setIndex)
+      ? attribute.setIndex === setIndex
+      : true;
+    if (attribute.semantic === semantic && matchesSetIndex) {
+      return attribute;
+    }
+  }
+};
+
+/**
+ * Create a bounding sphere from a primitive's POSITION attribute and model
+ * matrix.
+ *
+ * @param {ModelComponents.Primitive} primitive The primitive components.
+ * @param {Matrix4} modelMatrix The primitive's model matrix.
+ */
+ModelExperimentalUtility.createBoundingSphere = function (
```

Comment:
@sanjeetsuhag Testing out a model on another branch, I found a case where the bounding sphere isn't being computed correctly:
![image](https://user-images.githubusercontent.com/8422414/128208359-8ce0a514-7476-402a-aaa0-14359c017c90.png)


the bounding spheres should look like this: (old Model)
![image](https://user-images.githubusercontent.com/8422414/128208311-3a51af80-03cd-4848-a1d9-7dc660a69f66.png)

[Sandcastle with debug volumes](http://localhost:8080/Apps/Sandcastle/index.html#c=pVh7c9u4Ef8qqP6J1JEhEiRF0q/p2U5yaZ3GEzuZ6Z1uHIiERNQkwQFB2+qNv3sX4NvSXe5aj0ciFot9/PbBhR6pRI+cPTGJzlDOntAlK3mV4a+GNl1NIrO+FLmiPGdyNZmdrPJVzjdo+pf6IC4jljNcFiyqUirf5o9cijxjufpIi/K2KgohFYtn6NdVjtATz2PxhGnKpJpqAkKryV3CS7SW4qkEO2LBSpQLhcr6KGolI9aLRhnIxquJlqAtetFGPY55QP2Xz9foTPOsJolSRXm8WNQO4Uhki6Japzxa3NI8jmipUnZLsyJlV1TRxQOnkt4v7+lGMZkLkd+Th3u+TvGDeiaryUmtLRMxS40S0IDxAv4HMsxuubihT3nzibfp2hxe5YsFMm6/cgo90RIVUoCZJYtRVfJ8i/6RAIco35Rom969Qx8urlGtRmJ0J1p2tBOVROIpPzbSbXSErmCVChojcBGtK57GSCWslbcnDU1blLZcJdXaoNQwv5eiKhb6xBGcOGpOzLDRRUDX5ypH36KUoyOeF5W6oSpBiwI+F0oseEa3DCexREeiUpfVmkF0+m2gwRGD7Tc8kJR8QxshEU1TJArFBSDYYifQFtJOUsWMS2WRMMkjmqKEykzkPEKRYJsNjzggC9nFUvE0BzwZei/ENmUA5jueUoM7APhvFqkDuGnRHZtkKaMg4BBKWyN0sWl4Fw1vuQdQBmajoyO1K9gZ+AuPMStSsTt7BcYB7ACaT2CQ1JhkFJyikgE2pWhLhcV4rCVhaXEIQrDIAYvu6EONXgtlPAZNZ00LQ4Z43nSHv98iSNIyAZBqXHFbfdeWNe4jlxSsKjnNHVPsNiaO61vhkliu7/nLcN5Qie0ubRIurcD2XKel+o7tLYPAdhzihP4qnzVld23f29/RYzteSGybWEvXWdrLRqLt28T2/TC0g6Vv+a2ewF+6ru86oR06vjPU8113XE+fscNg6QQhmXdUD1T4QRBaoKtTQwIghi7xLDcIhmp+35sjMNxaho7lO8QnvuXVEjXZJh4hLgj2iWWFHdlzwKLACYjjhO5QE7kn31NlBUtClqFr2Z5nkU6m5YcQjtC1CXFs1+tUWXZoEzcMAD/AdqTquzEKwgDSwXZCxwu8Fjw78EILog5+um6fC7brAI9vWz5xRuCR34+RhS2IeGC5HqTXssk4IBLLhlh7rsmxjgjAOd4yJA4Ea+SM/QdgCwA2YnlW6DhBB1sAiW2DBeCL6wV94CA1XQsSgoSeP3LnOwECKz3i2m7gQoisBjVNdQPHAsUW1BBkX0+2oaog6S0vWHphr2hU6GfoZyjduSks/Wke9RPki/k0z5pKzBP55aQt+YTxbaJAAuhqRCfFq2niR0ZjeI/dcBUln0WaToF3jtqP1iIh+RZ6zNm+53gjRXbFtpKxssGbOJCRLmBbe+q6AIvl+FYdytqo3lnzJv5IleTPvfw7SfNSt9ISJ68MvBPv+DOL30lo5kZhbVstu5C14IFoEDoahgrJM674IysxjWMjoVH6UbMbf96najP9tR6BKpked8PEvKYNbD4eLtptnvOsym7AzvSW/4cdI5sEzV7M1tX2Fvrzhahy7dhXkVYZsChZMc3yMmtdMIKxBP93N2AUL80+hoafTzdVHuk3BpoarmaCQ8ikDyAjae93vTYiNQu8YH4SIkNK1KYPDsIoKQFiM3GOJ8gI4pvfFjRil0baZcd60p+X9UCn/wi20F8RoJLgjD7XRuJ14/KtHggYlhDYqpw39gLwVakgDDmjctZI7S3CDaja9CteKppHDKyUoMXCXueccQPqxrjQxtXExsVZlSpepLuL3Y3geTviNtHE+2Fsd16ZXYvveA73g1m9PRug0yTyyK4EK/EZyJDtU+I0BdceKHTK/yb7ERmwNxCmQjz80Dv225aOqp7mWzZtrJvXWucGWAjiKz9Geroinb4C+sPV23/efbj716wLSw1kNwj+2MyBl+NGN+x7J+ODBy8wcGTvTjFM8zsYn3Qam8KGAqPRQ1lPpErPpmKD9FSuEwvRooD7BtUlhfsI9IfP2gJDCMqV5zr9flA/sRzGzKYF4L2NBvmXk84iKJtHuF0ZI3rhGcvWTJYwxUFNPuQieoA5E4k1XLke6TplZWNSA3PLgY1H007ObOj7BVzmXqkB4Zpw9ekjggk4M6ADBpr25QN8UQXxhVlUz6RcDWBQQqRrqisqFlGlD+ItU29rGRe7D9BFJw1PfQk9ZKxGeKetgiwre6PnrfTe+q63ldUaGg9fsxuquz3k8jSH71kfi1dKWjJ0STDwUwfgUJ2RMGDslAx6KlTKV5pWQ01dMv6sBfxSv0QN00nP8tI6/9L6csCF1WQvUdqruz7RX3n1++lObOHuclEpJfKurleTL3DVMVfLuJJwrxpeVFP9egWMV5Ou8DdwEWH9qnMzSlj00F3/6z/z68Ghjf+jiMfH/0wpd8BCxoLL/4M50LrZhucs/lO2HDr10j6+DHvii8klLPTl7wle0YMsYlIK2WE4+nWl3moEwNdkPjkt1S5l562Sv/HM/MIC48cU44VicKeHplUu1hWERuGoLNtkO10Mj57G/BHx+OzA70MoSmlZws6mSs1gspqcny6Af++ovllCDn2CZpXSnWZL7PPrmogxPl3A8vDJrg90npwqXYLnPZCnai3i3fkwHqdKjtaaEp9ft3WCoDXVlXK6APo+55gCNPM7BzI3+dVE6jccuA/SYAWvTf1Mn+GZmOdSsaLesGEVU0WP1hAtID3qAj/eb/lzZHa+FMAM+2+MujcDpw8botiz0voAeli5f1DZntg9EIAwxA+WI4RhPQjBMG7/BQ)

Might be good to check `ModelUtility.computeBouningSphere` -- it's doing something more involved with the scene graph

Reply:
![image](https://user-images.githubusercontent.com/5172619/128401705-35fce819-623f-4d0c-99f8-ac3c6a4c87d6.png)

Fixed in the latest version.